### PR TITLE
Doc timer docs

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Minor documentation and style changes in TimerBase and friends.

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -1267,10 +1267,10 @@ class TimerBase(object):
         Boolean flag indicating whether this timer should operate as single
         shot (run once and then stop). Defaults to `False`.
 
-    callbacks : list
-        Stores list of (func, args) tuples that will be called upon timer
-        events. This list can be manipulated directly, or the functions
-        `add_callback` and `remove_callback` can be used.
+    callbacks : List[Tuple[callable, Tuple, Dict]]
+        Stores list of (func, args, kwargs) tuples that will be called upon
```
Comment (jklymak) [inline lib/matplotlib/backend_bases.py]: I can guess what these tuples may look like, but it'd be nice if there were an example, or there was a link to where I could see an example....

File: lib/matplotlib/backend_bases.py [hunk 2]
```diff
@@ -2405,10 +2408,13 @@ def new_timer(self, *args, **kwargs):
         ----------------
         interval : scalar
             Timer interval in milliseconds
-        callbacks : list
+        callbacks : List[Tuple[callable, Tuple, Dict]]
             Sequence of (func, args, kwargs) where ``func(*args, **kwargs)``
```
Comment (jklymak) [inline lib/matplotlib/backend_bases.py]: as above.  Though I have to admit it is likely my lack of deep python knowledge showing through here.

Comment (tacaswell) [reply lib/matplotlib/backend_bases.py]: Is the un-changed line below this clear? 

Are you asking about the inner tuple or the outer tuple?

See https://github.com/matplotlib/matplotlib/issues/10183#issuecomment-355793465 for this used in the wild.

Comment (jklymak) [reply lib/matplotlib/backend_bases.py]: The unchanged line.  I don't know how to represent `args` in the tuple.  Is it a tuple of the args?  A list? i.e. `ax.plot(x, y, linewidth=2)` would I write as `(ax.plot, (x, y), {'linewidths': 2})`?  

As I said, its probably just my unfamiliarity with what `*args` and `**kwargs` really mean.  They don't get used in a lot of other places in the language.

Comment (tacaswell) [reply lib/matplotlib/backend_bases.py]: https://docs.python.org/3/tutorial/controlflow.html#keyword-arguments and the next we sections (4.7.2 - 4.7.4) .  Basically in the function definition side the `*` and `**` in the signature mean 'collect all the positional arguments that don't have a named position and put them in a tuple' and 'collect all of the keywords which are explicitly listed and put them in a dict'.  On the call side, `*` and `**` mean 'unpack' so `foo(1, 2)` is the same as `foo(*(1, 2))` (technically works with any iterable) and `**` is the same for dictionaries `foo(a=1, b=2)`  is the same as `foo(**{'a':1, 'b':2})`.

This is super useful when you want to write generic wrappers ex

```python

def my_decorator(func):
    def inner(*args, **kwargs):
         print("I'm a decorator")
         return func(*args, **kwargs)
    return inner
```

or you are doing callbacks / defered execution

 ```python
call_later = [(f, (1, 2), {'a': 3}), (f2, (), {}), (f3, ('foo',), {})]
# some time later
for func, args, kwargs in call_later:
     func(*args, **kwargs)
```

We also use it through mpl to 'forward' kwargs down to the Artist class's `__init__` methods and then up the MRO chain.

Comment (jklymak) [reply lib/matplotlib/backend_bases.py]: Hi @tacaswell, sorry, I didn't mean to squeeze you for a python lesson ;-) I *have* looked that stuff up before, and get it.  I simply meant that I wouldn't have been sure about the `call_later` syntax in your example above off the top of my head.

We can argue how pandering the docs should be, but in this case, it'd be nice if there were something like:

```python
"""
i.e. `callbacks = [(func1, (1, 2), {'a': 3}), (func2, (), {}), (func3, ('foo',), {})]`
"""
```

to help the less with-it.

**Pull request merged**